### PR TITLE
Disable remove first ListField element

### PR DESCRIPTION
**Pull request by ksobolewski-netins**

### CODE REVIEW

**Comment by mprzodala** [comment]
I dont know why you give only posibility to disable remove first list element. Please change this logic and give someone posibility to declarate minimal lenght and maximum lenght of list.

**Comment by ksobolewski-netins** [comment]
Remove first element disallow is the most popular case for that action type(demo example: book author >= 1) that's why I created it. I'll change it with your suggestions.

**Comment by mprzodala** [comment]
ok thanks. I waiting for your commit.

File: src/components/ListField.jsx [hunk 1]
```diff
@@ -112,6 +116,26 @@ export class ListField extends React.Component {
         this.props.onChange(model.map(item => item.value));
     }
 
+    isAddAllowed() {
```
Comment (mprzodala) [inline src/components/ListField.jsx:115]: I think we should change this a little bit.
```js
isAddAllowed() {
    const { maxLength } = this.props;
    const { model } = this.state;
    if (typeof maxLength === 'number') return model.length < maxLength;
    return true;
}
```

File: src/components/ListField.jsx [hunk 2]
```diff
@@ -4,12 +4,16 @@ import FieldConnect from './FieldConnect';
...
     static defaultProps = {
-        value: []
+        value: [],
     };
 
     static generateItemId() {
         return Math.random().toString(36).substring(7);
     }
+
+    static isNaturalNumber(number) {
```
Comment (mprzodala) [inline src/components/ListField.jsx]: We don't need this method.

File: src/components/ListField.jsx [hunk 3]
```diff
@@ -112,6 +116,26 @@ export class ListField extends React.Component {
...
+        const model = Array.from(this.state.model);
+        if (newMaxLength && ListField.isNaturalNumber(newMaxLength)) {
+            if (minLength && ListField.isNaturalNumber(minLength) && newMaxLength < minLength)
+                newMaxLength = minLength;
+            return model.length < newMaxLength;
+        }
+        return true;
+    }
+
+    isRemoveAllowed() {
```
Comment (mprzodala) [inline src/components/ListField.jsx]: I think we should change this a little bit.
```js
isRemoveAllowed() {
    const { minLength } = this.props;
    const { model } = this.state;
    if (typeof minLength === 'number') return model.length > minLength;
    return true;
}
```

File: src/components/ListField.jsx [hunk 4]
```diff
@@ -112,6 +116,26 @@ export class ListField extends React.Component {
         this.props.onChange(model.map(item => item.value));
     }
 
+    isAddAllowed() {
+        const { minLength, maxLength } = this.props;
```
Comment (mprzodala) [inline src/components/ListField.jsx]: please add this props to propTypes

File: tests/ListField.test.jsx [hunk 5]
```diff
@@ -97,4 +102,103 @@ describe('ListField', () => {
...
+        expect(wrapper.find('input').length).toBe(1);
+
+        wrapper.find('.addButtonClass').first().simulate('click');
+
+        expect(wrapper.find('.addButtonClass').length).toBe(1);
+        expect(wrapper.find('.removeButtonClass').length).toBe(2);
+        expect(wrapper.find('input').length).toBe(2);
+    });
+
+    it("should set maxLength as minLength when it's smaller", () => {
```
Comment (mprzodala) [inline tests/ListField.test.jsx]: why ?

**Review by mprzodala — CHANGES_REQUESTED** [review]
CR done please check comments

**Review by mprzodala — APPROVED** [review]

**Comment by ksobolewski-netins** [comment]
Thanks for support and quick response 👍

**Pull request merged**